plugins/woocommerce/includes/admin/settings/class-wc-settings-accounts.php

```diff
@@ -131,7 +131,7 @@ protected function get_settings_for_default_section() {
 			),
 			array(
 				'title'             => __( 'Account creation options', 'woocommerce' ),
-				'desc'              => __( 'Use email address as account login (recommended)', 'woocommerce' ),
+				'desc'              => __( 'Create username from customer name (recommended)', 'woocommerce' ),
```

Comment:
This still isn't entirely correct. If there isn't a first name and last name, the logic will [fallback to using the email](https://github.com/woocommerce/woocommerce/blob/57c46969d62cee32fdb897a6d90673175b5661a1/plugins/woocommerce/includes/wc-user-functions.php#L228).

Reply:
@nerrad i updated it to the following now. "Use first and/or last name if available and valid to create the username. If neither is usable, fall back to the email address (recommended)." i think that should cover everything and be as clear as possible. what do you think? feel free to edit / suggest as per your suggestion